pkg/clustermesh/common/remote_cluster.go

```diff
@@ -126,6 +147,34 @@ func (rc *remoteCluster) releaseOldConnection() {
 }
 
 func (rc *remoteCluster) restartRemoteConnection() {
+	rc.mutex.Lock()
+	if rc.cacheTTL > 0 && rc.cacheRevokeCancel == nil {
+		rc.logger.Warn("Starting remote cluster cache TTL timer", logfields.TTL, rc.cacheTTL)
+
+		ctx, cancel := context.WithCancel(context.Background())
+		rc.cacheRevokeCancel = cancel
+
+		rc.wg.Add(1)
+		go func() {
+			defer rc.wg.Done()
```

Comment:
nit: you can now use waitgroups like this:
```suggestion
		rc.wg.Go(func() {
```